library/src/utils/serialization.ts

```diff
@@ -10,17 +10,31 @@ export function isSerializedUint8Array(data: unknown): data is BytesMessage {
   return type === 'bytes' && typeof value === 'string'
 }
 
-export function uint8ArrayToString(bytes: Uint8Array): string {
-  return new TextDecoder().decode(bytes)
+function convertBlobToBase64(blob: Blob): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader()
+    reader.readAsDataURL(blob)
+    reader.onloadend = () => {
+      const base64data = reader.result
+      resolve(base64data as string)
+    }
+    reader.onerror = error => {
+      reject(error)
+    }
+  })
+}
+
+export function uint8ArrayToString(bytes: Uint8Array): Promise<string> {
+  return convertBlobToBase64(new Blob([bytes]))
 }
 
 export function stringToUint8Array(serializedBytes: string): Uint8Array {
-  return new TextEncoder().encode(serializedBytes)
+  return Uint8Array.from(atob(serializedBytes.substring(37)), c => c.charCodeAt(0))
```

Comment:
Is not clear what exactly 37 means. Would be great to move to a constant with a self-explanatory name. 

If a certain line is expected there that needs to be removed, then it would be nice to check if it is really there. If there is not such a line as expected, then the data will be corrupted.